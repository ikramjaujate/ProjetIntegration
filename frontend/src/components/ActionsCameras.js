import '../css/Grades.css';


const ActionsCameras = ({name, allowed, notification}) => {

    /**
     * Ajout d'une bordure autour de la couleur sélectionnée
     * 
     * @author Clémentine Sacré <c.sacre@students.ephec.be>
     * @param {integer} idColor  Identifiant de la couleur sélectionnée
     */
     const cc = () => {
        
        console.log("oui");
    }


    return (
        <div className="row p-1 m-2 bg-light rounded col-9 col-sm-8 col-md-9 col-lg-7 col-xl-7 col-xxl-7">
            <div className="align-self-center col-6 col-sm-6 col-md-6 col-lg-6 col-xl-6 col-xxl-6">{name}</div>
            <div className="align-self-center col-3 col-sm-3 col-md-3 col-lg-3 col-xl-3 col-xxl-3"> 
                <div class="form-check form-switch">
                    <input class="form-check-input" type="checkbox" role="switch" checked={allowed ? true : false}/>
                </div>
            </div>
            <div className="rounded bg-notification col-2 col-sm-2 col-md-2 col-lg-2 col-xl-2 col-xxl-2">  
                {/* <i className={`bi ${notification ? "bi-bell-fill" : "bi-bell-slash-fill"}`} style={{color:notification ? "white" : allowed ? "var(--camera-allow)" : "var(--camera-refuse)"}}></i> */}
                <i type="button" className={`bi ${notification ? "bi-bell-fill" : "bi-bell-slash-fill"}`} style={{color:notification ? "var(--notification-on)" : "var(--notification-off)"}} onClick={() => cc()}></i>
            </div>
        </div>
    );
}

export default ActionsCameras;