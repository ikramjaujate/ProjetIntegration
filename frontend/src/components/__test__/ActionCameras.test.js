import React from 'react' ;
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16' ;
import 'jest-enzyme' ;
import 'jest-styled-components';
import ActionsCameras from '../ActionsCameras.js' ;
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });



describe('ActionCameras', () => {
    it('Snapshot - renders correctement', () => {
      const snapshot = renderer.create(<ActionsCameras allowed={false} id_camera="1" currentIdGrade="1"/>).toJSON();
      expect(snapshot).toMatchSnapshot();
    });

    it('Vérifier le texte affiché dans le titre', () => {
      const wrapper = shallow(<ActionsCameras name_camera="ZoneSud" />);
      const nameCamera = wrapper.find('.name-camera-grade');
      //expect(paragraph).toHaveLength(1);
      expect(nameCamera.text()).toEqual('ZoneSud');
    })

    it("Vérifier que le switch représentant l'action de la caméra est activé", () => {
      const wrapper = shallow(<ActionsCameras allowed={true} />);
      const check = wrapper.find('.form-check-input');
      expect(check.at(0).props().defaultChecked).toEqual(true);
    })

    it("Vérifier que le switch représentant l'action de la caméra est désactivé", () => {
      const wrapper = shallow(<ActionsCameras allowed={false} />);
      const check = wrapper.find('.form-check-input');
      expect(check.at(0).props().defaultChecked).toEqual();
    })

    it("Vérifier la classe du switch représentant l'action de la caméra", () => {
      const wrapper = shallow(<ActionsCameras id_camera="2" currentIdGrade="1" />);
      const check = wrapper.find('.form-check-input');
      expect(check.hasClass('action-1-2')).toEqual(true) ;
    })

    it("Vérifier l'id de la notification", () => {
      const wrapper = shallow(<ActionsCameras id_camera="2" currentIdGrade="1" />);
      const notification = wrapper.find('.bi');
      expect(notification.at(0).props().id).toEqual("notification-1-2") ;
    })


    it('Vérifier le texte affiché dans le titre a une valeur par défaut', () => {
      const wrapper = shallow(<ActionsCameras />);
      const nameCamera = wrapper.find('.name-camera-grade');
      expect(nameCamera.text()).toEqual('Chargement');
    })

    it("Vérifier la valeur par défaut du switch représentant l'action de la caméra", () => {
      const wrapper = shallow(<ActionsCameras />);
      const check = wrapper.find('.form-check-input');
      expect(check.at(0).props().defaultChecked).toEqual();
    })

    it("Vérifier la classe par défaut  du switch représentant l'action de la caméra", () => {
      const wrapper = shallow(<ActionsCameras />);
      const check = wrapper.find('.form-check-input');
      expect(check.hasClass('action-0-0')).toEqual(true) ;
    })

    it("Vérifier l'id par défaut de la notification", () => {
      const wrapper = shallow(<ActionsCameras />);
      const notification = wrapper.find('.bi');
      expect(notification.at(0).props().id).toEqual("notification-0-0") ;
    })

    it("Vérifie qu'il y a bien un fonction par défaut pour le changeAction", () => {
       const wrapper = shallow(<ActionsCameras />);
       const fctCheck = wrapper.find('.form-check-input');
       fctCheck.simulate('change') ;
    });

    it("Vérifie qu'il y a bien un fonction par défaut pour le changeNotification", () => {
      const wrapper = shallow(<ActionsCameras />);
      const fctNotification = wrapper.find('.bi-bell-slash-fill');
      fctNotification.simulate('click') ;
   });
    // it('`button` should be disabled', () => {
    //   const wrapper = shallow(<ActionsCameras name_camera="" id_camera="" notification="" changeAction="" changeNotification="" allowed={false} currentIdGrade="" />);
    //   const fctCheck = wrapper.find('.form-check-input');
    //   console.log("avant : ", fctCheck.at(0).props())
    //   fctCheck.simulate('change') ;
    //   console.log("apres : ",fctCheck.at(0).props().checked )
    //   expect(fctCheck.at(0).props().ischecked).toEqual(true);
    // });
})
  

  