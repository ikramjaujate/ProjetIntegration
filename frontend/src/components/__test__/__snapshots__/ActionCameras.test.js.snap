// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCameras Snapshot - renders correctement 1`] = `
<div
  className="layout-actions-cameras row p-1 m-2 bg-light rounded col-12 col-sm-12 col-md-12 col-lg-12 col-xl-12 col-xxl-12"
>
  <div
    className="name-camera-grade col-7 col-sm-7 col-md-7 col-lg-7 col-xl-7 col-xxl-7"
  >
    Chargement
  </div>
  <div
    className="switch-camera-grade col-3 col-sm-3 col-md-3 col-lg-3 col-xl-3 col-xxl-3"
  >
    <div
      className="form-check form-switch"
    >
      <input
        className="form-check-input input-switch action-1-1"
        onChange={[Function]}
        role="switch"
        type="checkbox"
      />
    </div>
  </div>
  <div
    className="rounded col-2 col-sm-2 col-md-2 col-lg-2 col-xl-2 col-xxl-2"
  >
    <i
      className="bi bi-bell-slash-fill"
      id="notification-1-1"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
